test/unit/BondingManager.js

```diff
@@ -1841,6 +1841,20 @@ contract("BondingManager", accounts => {
             assert.equal((await bondingManager.getDelegator(delegator1))[5], currentRound + 1, "should update caller's lastClaimRound to the current round")
         })
 
+        it("fires a ClaimEarnings event", async () => {
+            const expRewards = new BN(delegatorRewards * .3) // 30%
+            const expFees = new BN(delegatorFees * .3) // 30%
+            bondingManager.EarningsClaimed({from: delegator1}).on("data", e => {
```

Comment:
This event emitter based test isn't working properly because the test completes before the assertions in the callback are executed.

If this event emitter based test is replaced with a test that uses `truffleAssert`, the new test will fail because of the incorrect argument ordering when the `EarningsClaimed` event is fired:

```
const txResult = await bondingManager.claimEarnings(currentRound + 1, {from: delegator1})
            
truffleAssert.eventEmitted(txResult, "EarningsClaimed", e => {
    return e.delegate === transcoder &&
               e.delegator == delegator1 &&
               e.fees == expFees.toString() &&
               e.rewards == expRewards.toString() &&
               e.startRound == (currentRound + 1).toString() &&
               e.endRound == (currentRound + 1).toString()
})
```

Reply:
There's other tests in BondingManager.js that use the eventEmitter, would this be true for those tests as well then? Should we replace them all with `truffleAssert.eventEmitted` ?

Reply:
fixed in 0ee7707

Reply:
> There's other tests in BondingManager.js that use the eventEmitter, would this be true for those tests as well then?

Those might be a problem - let's take a look at those separately from this PR.

Reply:
I'll create an issue for it in the security audit prep project

Reply:
https://github.com/livepeer/protocol/issues/326